data/tcp-listener.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @title TCP Listener: Ping
+ * @difficulty intermediate
+ * @tags cli
+ * @run --allow-net <url>
+ * @resource {https://deno.land/api?s=Deno.listen} Deno.listen: API
+ *
+ * An example of a TCP listener on localhost that will log the message if written to and close the connection if connected to.
+ */
+const decoder = new TextDecoder();
```

Comment:
You should also add comments (like the other examples) explaining what's happening in the code, ex why do you need this TextDecoder?